src/ts/state/helpers.ts

```diff
@@ -20,3 +28,65 @@ export const addMatchesToState = <TMatch extends IMatch>(
     decorations
   };
 };
+
+/**
+ * Should we filter our decorations with the filterMatches predicate?
+ */
+export const shouldFilterDecorations = <TPluginState extends IPluginState>(
+  oldState: TPluginState,
+  newState: TPluginState,
+  filterMatches?: IFilterMatches<TPluginState["filterState"]>
+): filterMatches is IFilterMatches<TPluginState["filterState"]> => {
```

Comment:
I feel like this function could do with a test to confirm that it does indeed correctly answer the question.

Reply:
You're right – we verify it via the plugin behaviour, but I had a nagging feeling that it'd be better to test in isolation. Let's do that.